pysmt/smtlib/parser.py

```diff
@@ -66,7 +69,23 @@ def get_formula_fname(script_fname, environment=None, strict=True):
     """
     Returns the formula asserted at the end of the given script
     """
-    with io.BufferedReader(io.FileIO(script_fname, 'r')) as script:
+    if PY2:
```

Comment:
Lets try to keep the py2/py3 things outside the flow. We can introduce a thin wrapper to use in the with statement:

```
with Reader(script_fname) as script:
```

and define Reader depending from PY2/PY3